drivers/i2c/i2c_gecko.c

```diff
@@ -24,11 +25,16 @@ LOG_MODULE_REGISTER(i2c_gecko);
 #define DEV_BASE(dev) ((I2C_TypeDef *)((const struct i2c_gecko_config *const)(dev)->config)->base)
 
 struct i2c_gecko_config {
+#ifdef CONFIG_PINCTRL
+	const struct pinctrl_dev_config *pcfg;
+#endif
 	I2C_TypeDef *base;
 	CMU_Clock_TypeDef clock;
 	uint32_t bitrate;
+#ifndef CONFIG_PINCTRL
 	struct soc_gpio_pin pin_sda;
 	struct soc_gpio_pin pin_scl;
+#endif
```

Comment:
please, make pinctrl mandatory, old methods should actually be removed as deprecation period is gone.